### ISSUE TITLE
Jars required to compile from source? 

### ISSUE THREAD

**Issue by blaksec** [dependency-check/DependencyCheck#244]
Tried subscribe to groups but my request bounced back saying I have to be invited by a group owner, so I'm posting it here. 

I've been testing DependencyCheck for the past couple of days and it appears to be a nicely done, solid piece of code. The environment where I'm planning to put this requires everything to be built from source. I tried to get a list of dependencies from maven where it listed 91 dependencies, 21 of which do not exist in my environment which means I'll have to build them from scratch as well. It does appear that many of those dependencies are used during tests (older version of jetty, spring framework, daytrader app, etc), meaning I wont really need them.

Does anyone have a list of jars required just to compile the project source? Currently I only need 'core', 'util', 'cli', 'ant' components.

**Comment by jeremylong** [comment]
Look at the core [pom.xml line 428](https://github.com/jeremylong/DependencyCheck/blob/master/dependency-check-core/pom.xml#L428), all of the dependencies below that line are for testing. Basically, if you are not going to run the tests anything marked as test can be removed (the dependencies that are only used to analyze are always marked as test+optional).

Another way to get the list would be to run:

```
mvn org.apache.maven.plugins:maven-dependency-plugin:2.10:analyze-report
```

Anything marked optional is not used by the application. Does that work?

--Jeremy

**Comment by jeremylong** [comment]
Regarding the mailing list you should be able to just send emails to the
following addresses:

subscribe:     dependency-check+subscribe@googlegroups.com
unsubscribe: dependency-check+unsubscribe@googlegroups.com

This is documented on the projects mailing list page:
https://jeremylong.github.io/DependencyCheck/mail-lists.html

--Jeremy

**Comment by blaksec** [comment]
Looks like I was able to join the group this time. Also I was able to complile the project (mvn compile) after removing the dependencies below the line 428 with the exception of 'mailapi'. I will do a test run later today, compiling it without maven. If that works, perhaps pom.xml needs to be revisited and all the dependencies below 428 (except mailapi) have their scope changed from 'provided' to 'test'
